### DISCUSSION TITLE
ionospheric corrections in MintPy

### DISCUSSION THREAD

**Discussion by EJFielding** [insarlab/MintPy#1189] [Q&A] [answered]
Is the new ionospheric correction analysis in MintPy designed only for the ISCE2 topsStack (stackSentinel) ionospheric corrections, or does it support the alosStack, and stripmapStack ionospheric corrections?

**Comment by yunjunz** [comment, marked as answer]
We have tested it on alosStack and topsStack. But it should work for all isce2 stack processors. If that's not the case, please don't hesitate to open an issue and let us know.

**Comment by Qori88** [comment]
Hi Yunjun and Eric,
I implement correct ionosphere for alos1 from stackstripmap -W ionosphere. For mintpy.ionosphericDelay.method I use split_spectrum. But I found this error :

Estimate ionospheric delay time-series via ifgram_inversion.py ...
ifgram_inversion.py /media/imrsl/HDD2/TInSAR/Kalimantan/Pre/12_ion/mintpy3/inputs/ionStack.h5 -t /media/imrsl/HDD2/TInSAR/Kalimantan/Pre/12_ion/mintpy3/smallbaselineApp.cfg -w no --update
read input option from template file: /media/imrsl/HDD2/TInSAR/Kalimantan/Pre/12_ion/mintpy3/smallbaselineApp.cfg
phase unwrapping error correction "bridging+phase_closure" is turned ON
use dataset "unwrapPhase_bridging_phaseClosure" by default
Traceback (most recent call last):
  File "/home/imrsl/anaconda3/envs/mintpy/bin/smallbaselineApp.py", line 10, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/imrsl/anaconda3/envs/mintpy/lib/python3.11/site-packages/mintpy/cli/smallbaselineApp.py", line 209, in main
    run_smallbaselineApp(inps)
  File "/home/imrsl/anaconda3/envs/mintpy/lib/python3.11/site-packages/mintpy/smallbaselineApp.py", line 1155, in run_smallbaselineApp
    app.run(steps=inps.runSteps)
  File "/home/imrsl/anaconda3/envs/mintpy/lib/python3.11/site-packages/mintpy/smallbaselineApp.py", line 932, in run
    self.run_ionospheric_delay_correction(sname)
  File "/home/imrsl/anaconda3/envs/mintpy/lib/python3.11/site-packages/mintpy/smallbaselineApp.py", line 611, in run_ionospheric_delay_correction
    mintpy.cli.iono_split_spectrum.main(iargs)
  File "/home/imrsl/anaconda3/envs/mintpy/lib/python3.11/site-packages/mintpy/cli/iono_split_spectrum.py", line 145, in main
    run_iono_split_spectrum(inps)
  File "/home/imrsl/anaconda3/envs/mintpy/lib/python3.11/site-packages/mintpy/iono_split_spectrum.py", line 42, in run_iono_split_spectrum
    ifgram_inversion.main(cmd.split()[1:])
  File "/home/imrsl/anaconda3/envs/mintpy/lib/python3.11/site-packages/mintpy/cli/ifgram_inversion.py", line 264, in main
    inps = cmd_line_parse(iargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/imrsl/anaconda3/envs/mintpy/lib/python3.11/site-packages/mintpy/cli/ifgram_inversion.py", line 178, in cmd_line_parse
    raise ValueError(msg)
ValueError: input dataset name "unwrapPhase_bridging_phaseClosure" not found in file: /media/imrsl/HDD2/TInSAR/Kalimantan/Pre/12_ion/mintpy3/inputs/ionStack.h5

any guidance, I appreciate

**Reply by yunjunz** [reply]
@Qori88 could you add the following 

```python
    cmd += ' --dset unwrapPhase --mask-dset no '
```

after https://github.com/insarlab/MintPy/blob/d2e551a5a158fcac1ca518deae332101098c397d/src/mintpy/iono_split_spectrum.py#L40 and see if it fix the issue?

BTW, I am not sure if your input iono files are correct or not (have not tried iono correction within stripmapStack for many years), please ensure.

```cfg
##---------ionosphere stack (optional):
mintpy.load.ionUnwFile      = /media/imrsl/HDD2/TInSAR/Kalimantan/Pre/12_ion/Ionosphere/*_*/iono.bil.filt  #[path pattern of unwrapped interferogram files]
mintpy.load.ionCorFile      = /media/imrsl/HDD2/TInSAR/Kalimantan/Pre/12_ion/Ionosphere/*_*/iono.bil.unwCor.filt  #[path pattern of spatial coherence       files]
mintpy.load.ionConnCompFile = /media/imrsl/HDD2/TInSAR/Kalimantan/Pre/12_ion/Ionosphere/*_*/iono.bil.sig.filt  #[path pattern of connected components    files], optional but recommended
```

**Reply by Qori88** [reply]
Hi @yunjunz,
I add your code into estimate iono time-series part :
    # 3. estimate iono time-series
    print('\n'+'-'*80)
    print('Estimate ionospheric delay time-series via ifgram_inversion.py ...')
    cmd = f'ifgram_inversion.py {inps.iono_stack_file} -t {inps.template_file} -w no --update'
    cmd += ' --dset unwrapPhase --mask-dset no ' 
    print(cmd)
    ifgram_inversion.main(cmd.split()[1:])

And the error is changed to :
![Screenshot from 2024-07-25 21-15-06](https://github.com/user-attachments/assets/290aaef1-870f-4040-bb47-e2021f9ffbc1)

**Reply by Qori88** [reply]
> BTW, I am not sure if your input iono files are correct or not (have not tried iono correction within stripmapStack for many years), please ensure.

I also tried to change the ionosphere stack with interferogram stack that also have unw interferogram, spatial coh and conncomp but the error is the same

**Reply by yunjunz** [reply]
Could you try changing the following again?

from 

https://github.com/insarlab/MintPy/blob/d2e551a5a158fcac1ca518deae332101098c397d/src/mintpy/iono_split_spectrum.py#L40

to 

```python
    cmd = f'ifgram_inversion.py {inps.iono_stack_file} --dset unwrapPhase --weight-func no --update'
```

**Reply by Qori88** [reply]
Yunjun, it works with your last change suggestion. Thank you very much.
I continue to the last step but the final result is still messy. I will check my .cfg file again.

**Reply by yunjunz** [reply]
The fix is now in https://github.com/insarlab/MintPy/pull/1239

**Comment by Qori88** [comment]
here is my .txt file
[alos_pre12_ion.txt](https://github.com/user-attachments/files/16361708/alos_pre12_ion.txt)
